routing/http/server/server_test.go

```diff
@@ -73,20 +83,19 @@ func TestResponse(t *testing.T) {
 	runTest := func(t *testing.T, contentType string, expectedStream bool, expectedBody string) {
 		t.Parallel()
 
-		results := iter.FromSlice([]iter.Result[types.ProviderResponse]{
-			{Val: &types.ReadBitswapProviderRecord{
-				Protocol: "transport-bitswap",
-				Schema:   types.SchemaBitswap,
-				ID:       &pid,
-				Addrs:    []types.Multiaddr{},
-			}},
-			{Val: &types.ReadBitswapProviderRecord{
-				Protocol: "transport-bitswap",
-				Schema:   types.SchemaBitswap,
-				ID:       &pid2,
-				Addrs:    []types.Multiaddr{},
+		results := iter.FromSlice([]iter.Result[types.Record]{
+			{Val: &types.PeerRecord{
+				Schema:    types.SchemaPeer,
+				ID:        &pid,
+				Protocols: []string{"transport-bitswap"},
+				Addrs:     []types.Multiaddr{},
 			}},
```

Comment:
nit: in preexisting cases like this let's keep at least one with `Schema: types.SchemaBitswap,` because that is what https://cid.contact/routing/v1/providers/bafybeigdyrzt5sfp7udm7hu76uh7y26nf3efuylqabf3oclgtqy55fbzdi currently returns, and we should keep a regression test for it.

Reply:
I re-added it.